### PR TITLE
feat: added support for React Router 6

### LINKED ISSUE #259: Exception with react-router v6: (0, _reactRouterDom.withRouter)(LinkContainer) is not a function
I’m trying to use react-router-bootstrap with react-router v6 but I keep getting this error when the app starts.

```
LinkContainer.js:155 Uncaught TypeError: (0 , _reactRouterDom.withRouter) is not a function
    at Object../node_modules/react-router-bootstrap/lib/LinkContainer.js (LinkContainer.js:155)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object../node_modules/react-router-bootstrap/lib/IndexLinkContainer.js (IndexLinkContainer.js:14)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object../node_modules/react-router-bootstrap/lib/index.js (index.js:6)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Module../src/components/TopNavBar.js (TopNavBar.css?1017:45)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Module../src/app/App.js (App.css?56b3:45)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Module../src/index.js (index.css?f3f6:45)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object.1 (version.js:2)
    at __webpack_require__ (bootstrap:784)
    at checkDeferredModules (bootstrap:45)
    at Array.webpackJsonpCallback [as push] (bootstrap:32)
    at main.chunk.js:1
```

I can’t work out how to fix this – can anyone give me some suggestions?

I’m using versions:
```
    "react-router": "^6.0.0-beta.0",
    "react-router-bootstrap": "^0.25.0",
    "react-router-dom": "^6.0.0-beta.0",
```

Thanks,

     Ben.

### PR DESCRIPTION
**Pull request by mxschmitt**
Not sure if it was worth it.

There were quite a lot of changes  in React router 6: https://github.com/ReactTraining/react-router/blob/dev/docs/advanced-guides/migrating-5-to-6.md

Fixes #259

We would need a minor release after that since <6 seem to be not supported anymore.

### CODE REVIEW

File: src/LinkContainer.js [hunk 1]
```diff
@@ -1,46 +1,28 @@
...
+  activeClassName,
+  className,
+  activeStyle,
+  style,
+  isActive: getIsActive,
+  // eslint-disable-next-line comma-dangle
+  ...props
+}) => {
+  const navigate = useNavigate();
+  const href = useHref(typeof to === 'string' ? { pathname: to } : to);
```
Comment (kyletsang) [inline src/LinkContainer.js:24]: I think we can just pass `to` here as is?  Looks like `useHref` will parse both object and string forms.

**Comment by mxschmitt** [comment]
> I'm not well equipped to review this since i don't use react-router!

Then I would say let's wait if more users request that change and then decide to ship it.

**Comment by mxschmitt** [comment]
@jquense seems like #262 is one of the first users who face into that issue. Lets release a major version with v6 support or what's your opinion on that?

**Comment by mxschmitt** [comment]
@kyletsang anything else required to change in this PR or shall we follow-up with something?

**Comment by kyletsang** [comment]
I think this should be good.  I'll play around with it after it's merged and see if there's anything extra to add.  Thanks for making all these changes!

**Review by kyletsang — APPROVED** [review]

**Pull request merged**